drivers/timer/wch_systick_ch32v00x.c

```diff
@@ -27,20 +26,36 @@
 
 #define SYSTICK ((SysTick_Type *)(DT_INST_REG_ADDR(0)))
 
+static struct k_spinlock lock;
 static volatile uint32_t ch32v00x_systick_count;
 
 static void ch32v00x_systick_irq(const void *unused)
 {
 	ARG_UNUSED(unused);
 
+	k_spinlock_key_t key = k_spin_lock(&lock);
+
+	SYSTICK->CTLR &= ~(STK_STE);
 	SYSTICK->SR = 0;
+	SYSTICK->CNT = 0;
```

Comment:
Tell me more about the issue you're trying to fix? CNT was reset to zero by hardware, and then some time later this ISR runs, so you're loosing time.